File: backend/app/schemas/workflow.py
from datetime import datetime
from pydantic import BaseModel


class WorkflowStepCreate(BaseModel):
    name: str
    agent_id: str
    input_config: dict = {}
    order: int


class WorkflowCreate(BaseModel):
    name: str
    description: str | None = None
    steps: list[WorkflowStepCreate] = []


class WorkflowResponse(BaseModel):
    id: str
    name: str
    description: str | None
    status: str
    run_count: int
    created_at: datetime
    updated_at: datetime
    last_run_at: datetime | None

    model_config = {"from_attributes": True}

File: frontend/src/types/workflow.ts
export type WorkflowStatus = 'draft' | 'active' | 'paused' | 'completed' | 'failed'

export interface Workflow {
  id: string
  name: string
  description: string
  status: WorkflowStatus
  steps: WorkflowStep[]
  createdAt: string
  updatedAt: string
  lastRunAt: string | null
  runCount: number
}

export interface WorkflowStep {
  id: string
  name: string
  agentId: string
  input: Record<string, unknown>
  output: Record<string, unknown> | null
  status: 'pending' | 'running' | 'completed' | 'failed'
  order: number
}

export interface WorkflowRun {
  id: string
  workflowId: string
  status: 'running' | 'completed' | 'failed'
  startedAt: string
  completedAt: string | null
  steps: WorkflowStep[]
}
